src/components/FilterSearch.tsx

```diff
@@ -119,25 +125,35 @@ export function FilterSearch({
     const newDisplayName = itemData?.displayName as string;
     if (newFilter && newDisplayName) {
       if (select) {
-        if (currentFilter) {
-          searchActions.setFilterOption({ filter: currentFilter, selected: false });
-        }
-        searchActions.setFilterOption({ filter: newFilter, displayName: newDisplayName, selected: true
-        });
-        setCurrentFilter(newFilter);
-        setFilterQuery(newDisplayName);
-        executeFilterSearch(newDisplayName);
-        if (searchOnSelect) {
-          searchActions.setOffset(0);
-          searchActions.resetFacets();
-          executeSearch(searchActions);
+        if (onSelect && !searchOnSelect) {
```

Comment:
It feels weird that onSelect is only called when searchOnSelect is false. You are still *selecting* something even if searchOnSelect is true right?

Reply:
I think the idea was that if `searchOnSelect` is true, then the thing you want happening on select is the default "search" behavior. but, I talked with Davish about it again because the spec doesn't mention that (only the item description does), and I'm going to update to have both be allowed and independent as per his clarification

Reply:
The item description was based on an earlier conversation with Davish. We decided that `onSelect` would only be invoked if `searchOnSelect` was false. In an ideal world, I think there wouldn't be both `onSelect` and `searchOnSelect`. Instead, `onSelect` would be optional and the default behavior would be to fire off a search. Having both, and allowing them to be independent, feels odd to me. It's two props that allow you to dictate the behavior on select. 

I think the only way I'd feel good about having both and allowing them to be independent is if we marked `searchOnSelect` as deprecated. Internally, we'd refactor the Component so the default `onSelect` would be firing off a search. @oshi97 and @nmanu1 what do you think of that? We'd need to run it by Product too.

Reply:
I originally was also thinking of `onSelect` and `searchOnSelect` as both dictating how the selecting behavior should work. But, after thinking about it more and talking about it with Davish based on Oliver's comment, it made sense to me to have them be independent. Because `onSelect` dictates whether to use the default behavior or custom behavior when a filter is selected. But, `searchOnSelect` determines whether a searching action occurs or not after the filter selecting logic (whether it's default or custom) occurs. The user may want both a search to occur after a filter is selected as well as custom logic that happens when a filter is selected. In this case, a better name for `searchOnSelect` may be `searchAfterSelect`, although that may be confused with a boolean controlling the ordering of the searching and selecting actions. Does that make sense?

Reply:
My first intuition is that `onSelect` isn't related to whether or not FilterSearch runs a searchOnSelect. I think this was due to naming. If I saw the prop `onSelect` on FilterSearch I wouldn't *assume* that setting it would override the default behavior. 

If we want `onSelect` to override the default handler then maybe `handleSelect` would be a better name. In my brain, in general you can always have any amount of "onSomeEvent" listeners, but "handleSomeEvent" feels more like "this is the single designated handler". I'm probably splitting hairs.

Thinking about it a little more, though, having `handleSelect` override the default behavior (no `searchOnSelect`) lets us have less configuration. 
I think this is the ideal state.
Not because this is the most semantically clear, because that seems very subjective, but because that leads to the least configuration options (the least "entropy"). 

In order to prepare for that state, we'd want the props right now to behave in a way where we could remove `searchOnSelect` and no code would break. Right now, if we had `onSelect` NOT override the default behavior, in a even v2 we would have to tell people to change their `onSelect` handlers to also run a search at the start (or the end, this ordering business is also another reason we should stay away from `searchOnSelect`). Even though in a v2 we **can** make breaking changes, we shouldn't if we don't need to.

So anyways long story short after reading Tom's comment I'm kind of thinking we should have it override the default behavior? So we can mark `searchOnSelect` as deprecated, and if you have `handleSelect` set to something, that effectively sets `searchOnSelect` to false with higher priority.

Reply:
I thought about it some more and looked at the `SearchBar` component for comparison, I'm starting to agree with this approach. It makes sense that we would deprecate `searchOnSelect` because `onSelect` would allow the user to add that functionality if they wanted (although it would be harder than using `searchOnSelect`). I agree with Oliver that a better name for it though might `handleSelect` as opposed to `onSelect` since we're replacing the default behavior. But, right now in the `SearchBar`, we have `onSearch` and `onSelectVerticalLink` props that also overrides the default behavior instead of adding on to it. In that sense, `onSelect` is consistent.

Right now, the default behavior of `FilterSearch` is to have `searchOnSelect` as false. If we wanted to maintain that default and deprecate and eventually remove `searchOnSelect` (maybe in a v2), then eventually we would remove the code needed to conduct a search after selecting a filter. This code right now handles reseting the facets and offset before executing a search. My concern is that it may be unintuitive (and require some knowledge of business logic) for users to know that they would need to do those other things as part of executing a search to have it behave how they expect. If we're fine with changing the default behavior (i.e. most people want `searchOnSelect` to be true right now), then I think it makes sense to have default selecting and searching behavior (no `searchOnSelect` prop), which could be completely bypassed by custom logic in an `onSelect` prop.

To summarize:
- If product is fine with the change in the default behavior to have it include a search being executed, then I'll update the default value for `searchOnSelect` to true and mark it as deprecated.
- I'll keep the `onSelect` prop with the current naming (instead of `handleSelect`) unless we want to consider updating the naming in the `SearchBar` as well.
- I'll update `onSelect` to have higher priority than `searchOnSelect`. In other words, if `onSelect` is specified, it would completely bypass the selecting and searching logic, regardless of the value of `searchOnSelect`.
- If the user passes in an `onSelect` function and `searchOnSelect: true`, then we can log a console warning that the latter will be ignored.

Thoughts?

Reply:
I think those action items sound good! I do prefer the naming of `handleSelect` versus `onSelect`. But, it's not much of a preference. I'd be fine with either route in terms of naming. @oshi97 what do you think?

Reply:
yeah that sgtm!

Reply:
Heard back from Product and they're on board with deprecating the `searchOnSelect` prop now that we’re adding an `onSelect` prop. And, they said we can change the default value of `searchOnSelect` to true. I updated the PR accordingly